plugins/hls-cabal-plugin/src/Ide/Plugin/Cabal.hs

```diff
@@ -227,7 +231,14 @@ function invocation.
 kick :: Action ()
 kick = do
   files <- HashMap.keys <$> getCabalFilesOfInterestUntracked
+  Shake.ShakeExtras{ideTesting = Options.IdeTesting testing, lspEnv} <- Shake.getShakeExtras
+  let signal :: KnownSymbol s => Proxy s -> Action ()
+      signal msg = when testing $ liftIO $ Shake.mRunLspT lspEnv $
+        LSP.sendNotification (LSP.SMethod_CustomMethod msg) $
+        toJSON $ map fromNormalizedFilePath files
+  signal (Proxy @"kick/start/cabal")
   void $ uses Types.ParseCabalFile files
+  signal(Proxy @"kick/done/cabal")
```

Comment:
Can this wrapping functions be extracted and documented? While I remember what I did here, most other devs will not remember why they might need this.